src/privatesend/privatesend-client.cpp

```diff
@@ -1581,7 +1586,7 @@ bool CPrivateSendClientSession::CreateDenominated(CAmount nBalanceToDenominate,
         for (const auto it : mapDenomCount) {
             // Check if this specific denom could use another loop, check that there aren't nPrivateSendDenomsBatched of this
             // denom and that our nValueLeft/nBalanceToDenominate is enough to create one of these denoms, if so, loop again.
-            if (it.second < privateSendClient.nPrivateSendDenomsBatched && nValueLeft >= it.first && nBalanceToDenominate >= it.first) {
+            if (it.second < privateSendClient.nPrivateSendDenomsBatched && nValueLeft >= it.first && nBalanceToDenominate > 0) {
```

Comment:
why? I added this check intentionally to check that the denom we are checking is small enough to actually create a denom based on balance left to denominate. With this change, I think this could get caught in an infinite loop

Reply:
pls see 374f2fd commit message, the "final" denom takes care of a potential infinite loop and fixes an excessive creation of an additional tx

Reply:
I never experienced this excessive creation in my testing.. Is there a specific thing you have to do?